draco/analysis/sidereal.py

```diff
@@ -278,43 +285,122 @@ def process(self, sdata):
+            # Accumulate the variances in the stack.weight dataset.
+            self.stack.weight[:] += (coeff ** 2) * tools.invert_no_zero(sdata.weight[:])
+        else:
+            coeff = sdata.weight[:]
+            # We are using inverse variance weights.  In this case,
+            # we accumulate the inverse variances in the stack.weight
+            # dataset.  Do that directly to avoid an unneccessary
+            # division in the more general expression above.
+            self.stack.weight[:] += sdata.weight[:]
+
+        # Accumulate the total number of samples and the sum of squared coefficients.
+        self.stack.nsample[:] += count
+        self.sum_coeff_sq += coeff ** 2
+
+        # Below we will need to normalize by the current sum of coefficients.
+        # Can be found in the stack.nsample dataset for uniform case or
+        # the stack.weight dataset for inverse variance case.
+        if self.weight == "uniform":
+            sum_coeff = self.stack.nsample[:].astype(np.float32)
+        else:
+            sum_coeff = self.stack.weight[:]
+
+        # Calculate weighted difference between the new data and the current mean.
+        delta_before = coeff * (sdata.vis[:] - self.stack.vis[:])
+
+        # Update the mean.
+        self.stack.vis[:] += delta_before * tools.invert_no_zero(sum_coeff)
+
+        # Calculate the difference between the new data and the updated mean.
+        delta_after = sdata.vis[:] - self.stack.vis[:]
+
+        # Update the sample variance.
+        self.stack.sample_variance[0] += delta_before.real * delta_after.real
+        self.stack.sample_variance[1] += delta_before.real * delta_after.imag
+        self.stack.sample_variance[2] += delta_before.imag * delta_after.imag
+
     def process_finish(self):
-        """Construct and emit sidereal stack.
+        """Normalize the stack and return the result.
 
         Returns
         -------
         stack : containers.SiderealStream
             Stack of sidereal days.
         """
-
         self.stack.attrs["tag"] = "stack"
         self.stack.attrs["lsd"] = np.array(self.lsd_list)
 
-        self.stack.vis[:] *= tools.invert_no_zero(self.stack.weight[:])
+        nsample = self.stack.nsample[:]
+
+        # We need to normalize the sample variance by the sum of coefficients.
+        # Can be found in the stack.nsample dataset for uniform case
+        # or the stack.weight dataset for inverse variance case.
+        if self.weight == "uniform":
+            norm = nsample.astype(np.float32)
+            # Normalize the accumulated variances by the total number of samples
+            # and then invert to finalize stack.weight.
+            self.stack.weight[:] = (
+                tools.invert_no_zero(self.stack.weight[:]) * norm ** 2
+            )
+        else:
+            norm = self.stack.weight[:]
+
+        # Perform Bessel's correction.  In the case of
+        # uniform  weighting, norm will be equal to nsample - 1.
```

Comment:
Isn't this equal to `nsample - 1` only if each of the weights is 1 ? Otherwise they make a distinction between "frequency weights" and "reliability weights" here https://en.wikipedia.org/wiki/Weighted_arithmetic_mean#Weighted_sample_variance. This is probably not terribly important?